patches/server/0739-Add-more-Campfire-API.patch

```diff
@@ -0,0 +1,111 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: LemonCaramel <admin@caramel.moe>
+Date: Fri, 16 Jul 2021 00:39:03 +0900
+Subject: [PATCH] Add more Campfire API
+
+
+diff --git a/src/main/java/net/minecraft/world/level/block/entity/CampfireBlockEntity.java b/src/main/java/net/minecraft/world/level/block/entity/CampfireBlockEntity.java
+index 9f00de75b6f206c11ce7d1c59ba98fafe02fa9ab..d907d07ddfd6c608354a8251d177338f1b1ec6f1 100644
+--- a/src/main/java/net/minecraft/world/level/block/entity/CampfireBlockEntity.java
++++ b/src/main/java/net/minecraft/world/level/block/entity/CampfireBlockEntity.java
+@@ -33,12 +33,14 @@ public class CampfireBlockEntity extends BlockEntity implements Clearable {
+     private final NonNullList<ItemStack> items;
+     public final int[] cookingProgress;
+     public final int[] cookingTime;
++    public final boolean[] stopCooking; // Paper
+ 
+     public CampfireBlockEntity(BlockPos pos, BlockState state) {
+         super(BlockEntityType.CAMPFIRE, pos, state);
+         this.items = NonNullList.withSize(4, ItemStack.EMPTY);
+         this.cookingProgress = new int[4];
+         this.cookingTime = new int[4];
++        this.stopCooking = new boolean[4]; // Paper
+     }
+ 
+     public static void cookTick(Level world, BlockPos pos, BlockState state, CampfireBlockEntity campfire) {
+@@ -49,7 +51,9 @@ public class CampfireBlockEntity extends BlockEntity implements Clearable {
+ 
+             if (!itemstack.isEmpty()) {
+                 flag = true;
++                if (campfire.stopCooking[i]) { // Paper
```

Comment:
Am I reading this wrong, or should this be inverted, if not stopped, increment the cook progress

Reply:
omg.. i'll fix it